src/bika/lims/content/worksheet.py

```diff
@@ -135,8 +136,8 @@
 
     StringField(
         'ResultsLayout',
-        default='1',
-        vocabulary=WORKSHEET_LAYOUT_OPTIONS,
+        default=DEFAULT_WORKSHEET_LAYOUT,
+        vocabulary=getWorksheetLayouts(),
```

Comment:
Please add an upgrade step where you convert from the previous stored values of the database:

- `1` -> `analyses_classic_view`
- `2` -> `analyses_transposed_view`

Reply:
Maybe this block of code resolve problem?
```python
        view = self.context.getResultsLayout()
        if view not in self.layout_displaylist.keys():
            message = _("Layout view '{}' not found. Set '{}' as layout view.".format(view, DEFAULT_WORKSHEET_LAYOUT))
            self.context.plone_utils.addPortalMessage(message, "warning")
            view = DEFAULT_WORKSHEET_LAYOUT
            self.context.setResultsLayout(view)
```

This condition was added intentionally for the "first" check and subsequent possibly erroneous